pytorch/workflows/mnist.py

```diff
@@ -0,0 +1,181 @@
+
+from flytekit.sdk.tasks import pytorch_task
+
+WORLD_SIZE = int(os.environ.get('WORLD_SIZE', 1))
+
+
+class Net(nn.Module):
+    def __init__(self):
+        super(Net, self).__init__()
+        self.conv1 = nn.Conv2d(1, 20, 5, 1)
+        self.conv2 = nn.Conv2d(20, 50, 5, 1)
+        self.fc1 = nn.Linear(4*4*50, 500)
+        self.fc2 = nn.Linear(500, 10)
+
+    def forward(self, x):
+        x = F.relu(self.conv1(x))
+        x = F.max_pool2d(x, 2, 2)
+        x = F.relu(self.conv2(x))
+        x = F.max_pool2d(x, 2, 2)
+        x = x.view(-1, 4*4*50)
+        x = F.relu(self.fc1(x))
+        x = self.fc2(x)
+        return F.log_softmax(x, dim=1)
+
+def train(model, device, train_loader, optimizer, epoch, writer, log_interval):
+    model.train()
+    for batch_idx, (data, target) in enumerate(train_loader):
+        data, target = data.to(device), target.to(device)
+        optimizer.zero_grad()
+        output = model(data)
+        loss = F.nll_loss(output, target)
+        loss.backward()
+        optimizer.step()
+        if batch_idx % log_interval == 0:
+            print('Train Epoch: {} [{}/{} ({:.0f}%)]\tloss={:.4f}'.format(
+                epoch, batch_idx * len(data), len(train_loader.dataset),
+                       100. * batch_idx / len(train_loader), loss.item()))
+            niter = epoch * len(train_loader) + batch_idx
+            writer.add_scalar('loss', loss.item(), niter)
+
+def test(model, device, test_loader, writer, epoch):
+    model.eval()
+    test_loss = 0
+    correct = 0
+    with torch.no_grad():
+        for data, target in test_loader:
+            data, target = data.to(device), target.to(device)
+            output = model(data)
+            test_loss += F.nll_loss(output, target, reduction='sum').item() # sum up batch loss
+            pred = output.max(1, keepdim=True)[1] # get the index of the max log-probability
+            correct += pred.eq(target.view_as(pred)).sum().item()
+
+    test_loss /= len(test_loader.dataset)
+    print('\naccuracy={:.4f}\n'.format(float(correct) / len(test_loader.dataset)))
+    accuracy = float(correct) / len(test_loader.dataset)
+    writer.add_scalar('accuracy', accuracy, epoch)
+    return (epoch, accuracy)
+
+def epoch_step(model, device, train_loader, test_loader, optimizer, epoch, writer, log_interval):
+    train(model, device, train_loader, optimizer, epoch, writer, log_interval)
+    return test(model, device, test_loader, writer, epoch)
+
+def should_distribute():
+    return dist.is_available() and WORLD_SIZE > 1
+
+
+def is_distributed():
+    return dist.is_available() and dist.is_initialized()
+
+@inputs(
+    no_cuda=Types.Boolean,
+    batch_size=Types.Integer,
+    test_batch_size=Types.Integer,
+    epochs=Types.Integer,
+    learning_rate=Types.Float,
+    sgd_momentum=Types.Float,
+    seed=Types.Integer,
+    log_interval=Types.Integer,
+    save_model=Types.Boolean,
+    dir=Types.String)
+@outputs(out=Types.String)
```

Comment:
I think this needs some improvements like changing output to be the model. @wild-endeavor can you help?

Reply:
First of all MNIST in pytorch world is sort of 'hello world' (https://github.com/pytorch/examples/tree/master/mnist)

One can store model on their own by supplying `save_model=true`. Here we might only adjust the example to store on s3 (for now it is local FS).
Epoch - is model training iteration. Accuracy is acquired by evaluating test dataset on a trained model after each iteration.

I considered that accuracy is good enough to demonstrate, that job was done since 
 - it implies both training and evaluation steps
 - succinct result to show

Reply:
@igorvalko it is possible to output a model as a blob object in Flyte - Here is an example: The benefit is, you can then simply point your notebook to flyteadmin and load the model https://github.com/lyft/flytesnacks/blob/master/python/multi_step_linear/diabetes_xgboost.py#L104

Reply:
and lets rename out to accuracy?

Reply:
```suggestion
@outputs(accuracy=[Types.Float])
```

Reply:
done